lib/publiccloud/ssh_interactive.pm

```diff
@@ -52,7 +52,10 @@ sub ssh_interactive_tunnel {
 
     # Pipe the output of the device fifo to the local serial terminal
 # Note: We run this in a loop so that the ssh tunnel gets automatically re-established after device reboots and such. The sleep helps to avoid unnecessary CPU hogging in case of connection issues
-    enter_cmd("while true; do ssh sut -yt -R '$upload_port:$upload_host:$upload_port' 'rm -f /dev/sshserial && mkfifo -m a=rwx /dev/sshserial && tail -fn +1 /dev/sshserial' 2>&1 >/dev/$serialdev; sleep 5; done");
+    my $cmd = "while true; do ssh sut -yt -R '$upload_port:$upload_host:$upload_port' 'rm -f /dev/sshserial && mkfifo -m a=rwx /dev/sshserial && tail -fn +1 /dev/sshserial' 2>&1 >/dev/$serialdev; sleep 5; done";
+    # TODO: if not work try with nohup
+    script_run($cmd, timeout => 30);
```

Comment:
I don't see what this is improving. If I understand correctly, the tunnel command is expected to be running for the duration of the test. If run with `enter_cmd`, then it runs and we don't care for its return value. If run with `script_run`, `testapi` will add something like a `; echo $hash-$?` at the end of the command, which will not be printed because it's an endless loop, and then do a `wait_serial()` to try to get the return value. Since the return value is not printed, then the `wait_serial()` and kill the test due to a timeout ... see: https://github.com/os-autoinst/os-autoinst/blob/master/testapi.pm#L979

Reply:
Ok I try with a different approach then :+1: thanks for the feedback!